middleware.js

```diff
@@ -109,6 +97,21 @@ module.exports = function(options) {
 
   // Middleware
   return function sass(req, res, next) {
+    var sassMiddlewareError = null;
+
+    // This function will be called if something goes wrong
+    var error = function(err, errorMessage) {
+      if (debug) {
+        log('error', 'error', errorMessage || err);
+      }
+
+      if (options.error) {
+        options.error(err);
+      }
+
+      sassMiddlewareError = err;
+    };
+
```

Comment:
moved `error()` here so `sass()` is reentrant.

added logging of all errors if debugging is on